stdlib/public/SDK/Foundation/Codable.swift

```diff
@@ -0,0 +1,1606 @@
+    /// - precondition: May not be called after a previous `self.encode(_:)` call.
+    func encode(_ value: UInt64) throws
+
+    /// Encodes a single value of the given type.
+    ///
+    /// - parameter value: The value to encode.
+    /// - throws: `CocoaError.coderInvalidValue` if the given value is invalid in the current context for this format.
+    /// - precondition: May not be called after a previous `self.encode(_:)` call.
+    func encode(_ value: Float) throws
+
+    /// Encodes a single value of the given type.
+    ///
+    /// - parameter value: The value to encode.
+    /// - throws: `CocoaError.coderInvalidValue` if the given value is invalid in the current context for this format.
+    /// - precondition: May not be called after a previous `self.encode(_:)` call.
+    func encode(_ value: Double) throws
+
+    /// Encodes a single value of the given type.
+    ///
+    /// - parameter value: The value to encode.
+    /// - throws: `CocoaError.coderInvalidValue` if the given value is invalid in the current context for this format.
+    /// - precondition: May not be called after a previous `self.encode(_:)` call.
+    func encode(_ value: String) throws
+
+    /// Encodes a single value of the given type.
+    ///
+    /// - parameter value: The value to encode.
+    /// - throws: `CocoaError.coderInvalidValue` if the given value is invalid in the current context for this format.
+    /// - precondition: May not be called after a previous `self.encode(_:)` call.
+    func encode(_ value: Data) throws
+}
+
+/// A `SingleValueDecodingContainer` is a container which can support the storage and direct decoding of a single non-keyed value.
+public protocol SingleValueDecodingContainer {
+    /// Decodes a single value of the given type.
+    ///
+    /// - parameter type: The type to decode as.
+    /// - returns: A value of the requested type.
+    /// - throws: `CocoaError.coderTypeMismatch` if the encountered encoded value cannot be converted to the requested type.
+    func decode(_ type: Bool.Type) throws -> Bool
+
+    /// Decodes a single value of the given type.
+    ///
+    /// - parameter type: The type to decode as.
+    /// - returns: A value of the requested type.
+    /// - throws: `CocoaError.coderTypeMismatch` if the encountered encoded value cannot be converted to the requested type.
+    func decode(_ type: Int.Type) throws -> Int
+
+    /// Decodes a single value of the given type.
+    ///
+    /// - parameter type: The type to decode as.
+    /// - returns: A value of the requested type.
+    /// - throws: `CocoaError.coderTypeMismatch` if the encountered encoded value cannot be converted to the requested type.
+    func decode(_ type: Int8.Type) throws -> Int8
+
+    /// Decodes a single value of the given type.
+    ///
+    /// - parameter type: The type to decode as.
+    /// - returns: A value of the requested type.
+ 
```

Comment:
I think it's less readable to have dozens of copy and pasted versions of the same piece of code. What if you have to change all of them, but forget to change one place?